### ISSUE TITLE
[question] librdkafka is installed as 0.8.6 version. How can I have latest version of it?

### ISSUE THREAD

**Issue by kolinko-ant** [GoogleCloudPlatform/php-docker#487]
I use `rdkafka` extension and after installing it I have 0.8.6 version of `librdkafka` in the system. But the problem is this is old version and I need to have 1.1.0. Is there a way to say to GAE to install latest version of `librdkafka` library?

**Comment by jaitaiwan** [comment]
Hey @kolinko-ant I don't see librdkafka in the list of debian packages. How are you installing rdkafka?

**Comment by kolinko-ant** [comment]
Here is the package installer script https://github.com/GoogleCloudPlatform/php-docker/blob/master/package-builder/extensions/rdkafka/build.sh and command is `apt-get install -y librdkafka-dev` (the same way as in librdkafka docs https://github.com/edenhill/librdkafka#installing-prebuilt-packages).

**Comment by kolinko-ant** [comment]
I just specified `ext-rdkafka` in composer.json as docs says.

**Comment by jaitaiwan** [comment]
Weird. I completely missed it in the docs. Well we're on Ubuntu 16.04 https://github.com/GoogleCloudPlatform/php-docker/blob/master/php-base/Dockerfile

So it's likely an older version of the package is only what's available. Probably need to upgrade this entire repo to 18.04 or higher

**Comment by kolinko-ant** [comment]
@jaitaiwan I've asked the question about this here https://github.com/edenhill/librdkafka/issues/2488 and the solution is to use Confluent packages https://docs.confluent.io/current/installation/installing_cp/deb-ubuntu.html#systemd-ubuntu-debian-install to get latest version. 

Is that something that easily can be changed or it will take a lot of time to make this change?

**Comment by jaitaiwan** [comment]
Yeh that's easily changed BUT it won't make it to GAE unless the team merges the change upstream.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by kolinko-ant** [comment]
Sounds promising :). Would be nice to understand the timelines on when it might be merged to upstream? Then I can talk with my stakeholders and explain them for how much time we need to postpone kafka integration. Do you have any expectations when it can be done? Would really appreciate for the information.

**Comment by jaitaiwan** [comment]
I can't tell you sorry but I've been asking to become a maintainer... I'm the most active Dev on here at the moment.

If you have the time I can direct you to another PR that has instructions on how to make the changes and test it.

**Comment by kolinko-ant** [comment]
> If you have the time I can direct you to another PR that has instructions on how to make the changes and test it.

Would like to help. Maybe it will push things faster.

**Comment by kolinko-ant** [comment]
And just for complete understanding. Another way to make it work would be to use custom php runtime and change this docker config as much as we want. But I think it will take a lot of time to duplicate the same runtime and make it work. Cause we already in production. 

Any other options to have newer version of librdkafka that I don't see?

**Comment by jaitaiwan** [comment]
@kolinko-ant yeah it would definitely take a long time, but at this point the whole php-docker repo needs a re-write because we probably want to upgrade to ubuntu 18.04 or move to a slimmer debian image and squash all the layers it has so that it's a much smaller image for deployments.

No other options I can think of apart from rolling your own .deb file and running a custom docker build like you said.

The Pull Request I made to add phalcon support is here: #453 

You'll want to check the comments of the PR as the original maintainer gave instructions for how to run/test the updated library.

**Comment by kolinko-ant** [comment]
I am a bit confused now. If let's say I'll create a PR within 1-2 weeks, then is there a chance that this one gets merged to upstream and I will get it in GAE? Or if it will take a lot of time and you have a plans to upgrade ubuntu to 18.04 (which uses 0.11 version of librdkafka and looks like it's enough for me), maybe then it's not worth to invest time to this PR?

**Comment by jaitaiwan** [comment]
I have absolutely 0 visibility into how long it takes for things to make it
from this repo into runtime: php. If you do a simple PR for now to upgrade
the version of the lib that is probably the best bet that your change will
be accepted upstream because it's a relatively small change.
